pointcloud/basic_octree.go

```diff
@@ -258,9 +258,27 @@ func (octree *BasicOctree) Transform(pose spatialmath.Pose) spatialmath.Geometry
 }
 
 // ToProtobuf converts the octree to a Geometry proto message.
-// TODO (RSDK-3743): Implement BasicOctree Geometry functions.
 func (octree *BasicOctree) ToProtobuf() *commonpb.Geometry {
-	return nil
+	pc := NewBasicPointCloud(octree.Size())
+	octree.Iterate(0, 0, func(p r3.Vector, d Data) bool {
+		err := pc.Set(p, d)
+		return err == nil
+	})
```

Comment:
This should be unnecessary.  `ToBytes` takes a `PointCloud` which `BasicOctree` already satisfies the interface for